src/auth/registration/StepsButton.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  Box,
+  Button,
+  Stack,
+  Step,
+  StepLabel,
+  Stepper,
+  Typography,
+} from "@mui/material";
+import React, { useState } from "react";
+
+interface Props {
+  steps: {
+    label: string;
+    component: React.ReactNode;
+  }[];
+  finalComponent: React.ReactNode;
+}
+
+export function StepsButton({ steps, finalComponent }: Props) {
+  const [activeStep, setActiveStep] = useState(0);
+
+  return (
+    <Box sx={{ width: "100%" }}>
+      {steps[activeStep].component}
+      <Stepper activeStep={activeStep} sx={{ mt: 1 }}>
+        {steps.map((s) => {
+          const stepProps: { completed?: boolean } = {};
+          return (
+            <Step key={s.label} {...stepProps}>
+              <StepLabel>{s.label}</StepLabel>
+            </Step>
+          );
+        })}
+      </Stepper>
+      <Stack
+        direction="row"
+        spacing={1}
+        mt={2}
+        justifyContent="space-between"
+        alignItems="center"
+      >
+        {activeStep !== 0 && (
+          <Button
+            size="large"
+            variant="outlined"
+            onClick={(e) => {
+              e.preventDefault();
+              setActiveStep(activeStep - 1);
+            }}
+          >
+            Back
+          </Button>
+        )}
```

Comment:
nits:
- the first screen should have the next button on the right, not the left. (Either render an empty Box to force space-between to work, or render the back button in all cases but set `visibility: hidden`.)
- the back button should probably be `variant=text` rather than outlined